vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java

```diff
@@ -165,7 +283,8 @@ public void setPiece(GamePiece piece) {
 
       if (!contains) {
         try {
-          availableModel.addElement((GamePiece) pieceClass.getConstructor().newInstance());
+          addAdditionalElement((GamePiece) pieceClass.getConstructor().newInstance());
+          setSortOrder();
```

Comment:
I know this is awkward to change, but we don't want to catch anything thrown by `setSortOrder()` here, right?

Reply:
setSortOrder() shouldn't throw anything and if the addAdditionalElement() failed, then there is no need to reset the Sort order.

Reply:
My concern is that if `setSortOrder()` unexpectedly throws something, we'll never see it because it will get swallowed up by the `catch (Throwable t)` just below it. That's the broadest possible catch, so should enclose only the one line which could legitimately throw anything because it could be running the constructor of a custom class.

Reply:
Ah true. If setSortorder() ever throws something, we want a Bug report. I'll just move it down below the try.